Getting failures on MacOS.

Built with `./Tools/wasm/wasm_build.py wasi compile > wasi_compile.txt 2>&1` (`python` is Python 3.10): [wasi_compile.txt](https://github.com/python/cpython/files/9330209/wasi_compile.txt)

Testing with `./Tools/wasm/wasm_build.py wasi test`:
```
❯ ./Tools/wasm/wasm_build.py wasi test
make: realpath: Command not found
The following modules are *disabled* in configure script:
_sqlite3

The necessary bits to build these optional modules were not found:
_bz2                  _ctypes               _hashlib
_lzma                 _ssl                  _testimportmultiple
_testmultiphase       _uuid                 readline
xxlimited             xxlimited_35          zlib
To find the necessary bits, look in configure.ac and config.log.

Could not build the ssl module!
Python requires a OpenSSL 1.1.1 or newer

Checked 109 modules (75 built-in, 0 shared, 21 n/a on wasi-wasm32, 1 disabled, 12 missing, 0 failed on import)
make: realpath: Command not found
_PYTHON_HOSTRUNNER='wasmtime run --env PYTHONPATH=//build/lib.wasi-wasm32-3.12:/Lib --mapdir /::/Users/brettcannon/Repositories/python/cpython/builddir/wasi/../.. --' _PYTHON_PROJECT_BASE=/Users/brettcannon/Repositories/python/cpython/builddir/wasi _PYTHON_HOST_PLATFORM=wasi-wasm32 PYTHONPATH=/Users/brettcannon/Repositories/python/cpython/builddir/wasi/build/lib.wasi-wasm32-3.12:/Users/brettcannon/Repositories/python/cpython/builddir/wasi/../../Lib _PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata__wasi_wasm32-wasi /Users/brettcannon/Repositories/python/cpython/builddir/build/python.exe  /Users/brettcannon/Repositories/python/cpython/builddir/wasi/../../Tools/scripts/run_tests.py -j 1 -u all -W --slowest --fail-env-changed --timeout=1200 -i '*.test_srcdir' -j2
/Users/brettcannon/Repositories/python/cpython/builddir/build/python.exe -u -W default -bb -m test -r -w --python 'wasmtime run --env PYTHONPATH=//build/lib.wasi-wasm32-3.12:/Lib --mapdir /::/Users/brettcannon/Repositories/python/cpython/builddir/wasi/../.. -- python.wasm' -j 1 -u all -W --slowest --fail-env-changed --timeout=1200 -i '*.test_srcdir' -j2
Raised RLIMIT_NOFILE: 256 -> 1024
== CPython 3.12.0a0 (heads/wasm_build-dirty:66f6fd8897, Aug 12 2022, 15:51:21) [Clang 12.0.5 (clang-1205.0.22.9)]
== macOS-12.3.1-x86_64-i386-64bit little-endian
== cwd: /Users/brettcannon/Repositories/python/cpython/builddir/wasi/build/test_python_90919æ
== CPU count: 4
== encodings: locale=UTF-8, FS=utf-8
Using random seed 5054115
0:00:00 load avg: 7.71 Run tests in parallel using 2 child processes (timeout: 20 min, worker timeout: 25 min)
0:00:06 load avg: 7.73 [  1/435/1] test_uuid crashed (Exit code 1)
Could not find platform independent libraries <prefix>
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Lib/test/regrtest.py", line 11, in <module>
    from test.libregrtest import main
  File "/Lib/test/libregrtest/__init__.py", line 1, in <module>
    from test.libregrtest.cmdline import _parse_args, RESOURCE_NAMES, ALL_RESOURCES
  File "/Lib/test/libregrtest/cmdline.py", line 5, in <module>
    from test.support import os_helper
  File "/Lib/test/support/__init__.py", line 1229, in <module>
    not sysconfig.get_config_var('WITH_DOC_STRINGS'))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Lib/sysconfig.py", line 715, in get_config_var
    return get_config_vars().get(name)
           ^^^^^^^^^^^^^^^^^
  File "/Lib/sysconfig.py", line 670, in get_config_vars
    _init_posix(_CONFIG_VARS)
  File "/Lib/sysconfig.py", line 531, in _init_posix
    _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named '_sysconfigdata__wasi_wasm32-wasi'
```

Then that import failure repeats.

--- a/Tools/wasm/README.md
+++ b/Tools/wasm/README.md
@@ -51,7 +51,7 @@ make -j$(nproc)
 popd
 ```
 
-### Cross compile to wasm32-emscripten for browser
+### Cross-compile to wasm32-emscripten for browser
 
 ```shell
 ./Tools/wasm/wasm_build.py emscripten-browser

--- a/Tools/wasm/wasm_build.py
+++ b/Tools/wasm/wasm_build.py
@@ -1,5 +1,5 @@
 #!/usr/bin/env python3
-"""Build script for Python on WebAssembly platforms
+"""Build script for Python on WebAssembly platforms.
 
   $ ./Tools/wasm/wasm_builder.py emscripten-browser compile
   $ ./Tools/wasm/wasm_builder.py emscripten-node-dl test
@@ -342,7 +342,7 @@ class BuildProfile:
         env.update(self.host.platform.getenv())
         return env
 
-    def _run_cmd(self, cmd: List[str], args: List[str]):
+    def _run_cmd(self, cmd: Iterable[str], args: Iterable[str]):
         cmd = list(cmd)
         cmd.extend(args)
         return subprocess.check_call(

--- a/Tools/wasm/wasm_build.py
+++ b/Tools/wasm/wasm_build.py
@@ -24,7 +24,9 @@ import shlex
 import shutil
 import subprocess
 import sysconfig
-from typing import Callable, List, Union  # for Python 3.8
+
+# for Python 3.8
+from typing import Any, Dict, Callable, Iterable, List, Optional, Union
 
 SRCDIR = pathlib.Path(__file__).parent.parent.parent.absolute()
 WASMTOOLS = SRCDIR / "Tools" / "wasm"
@@ -77,7 +79,7 @@ https://wasmtime.dev/ to install wasmtime.
 """
 
 
-def get_emscripten_root(emconfig: pathlib.Path = EM_CONFIG) -> pathlib.Path:
+def get_emscripten_root(emconfig: pathlib.Path = EM_CONFIG) -> pathlib.PurePath:
     """Parse EM_CONFIG file and lookup EMSCRIPTEN_ROOT
 
     The ".emscripten" config file is a Python snippet that uses "EM_CONFIG"
@@ -89,7 +91,7 @@ def get_emscripten_root(emconfig: pathlib.Path = EM_CONFIG) -> pathlib.Path:
     with open(emconfig, encoding="utf-8") as f:
         code = f.read()
     # EM_CONFIG file is a Python snippet
-    local = {}
+    local: Dict[str, Any] = {}
     exec(code, globals(), local)
     return pathlib.Path(local["EMSCRIPTEN_ROOT"])
 
@@ -127,9 +129,9 @@ class Platform:
 
     name: str
     pythonexe: str
-    config_site: pathlib.Path
-    configure_wrapper: pathlib.Path
-    make_wrapper: pathlib.Path
+    config_site: Optional[pathlib.PurePath]
+    configure_wrapper: Optional[pathlib.PurePath]
+    make_wrapper: Optional[pathlib.PurePath]
     environ: dict
     check: Callable[[], None]
 
@@ -144,12 +146,13 @@ def _check_clean_src():
     ]
     for candidate in candidates:
         if candidate.exists():
-            raise DirtySourceDirectory(candidate, CLEAN_SRCDIR)
+            raise DirtySourceDirectory(os.fspath(candidate), CLEAN_SRCDIR)
 
 
 NATIVE = Platform(
     "native",
-    pythonexe=sysconfig.get_config_var("BUILDPYTHON"),  # macOS has python.exe
+    # macOS has python.exe
+    pythonexe=sysconfig.get_config_var("BUILDPYTHON") or "python",
     config_site=None,
     configure_wrapper=None,
     make_wrapper=None,
@@ -164,17 +167,17 @@ def _check_emscripten():
     # sanity check
     emconfigure = EMSCRIPTEN.configure_wrapper
     if not emconfigure.exists():
-        raise MissingDependency(emconfigure, INSTALL_EMSDK)
+        raise MissingDependency(os.fspath(emconfigure), INSTALL_EMSDK)
     # version check
     version_txt = EMSCRIPTEN_ROOT / "emscripten-version.txt"
     if not version_txt.exists():
-        raise MissingDependency(version_txt, INSTALL_EMSDK)
+        raise MissingDependency(os.fspath(version_txt), INSTALL_EMSDK)
     with open(version_txt) as f:
         version = f.read().strip().strip('"')
     version_tuple = tuple(int(v) for v in version.split("."))
     if version_tuple < EMSDK_MIN_VERSION:
         raise MissingDependency(
-            version_txt,
+            os.fspath(version_txt),
             f"Emscripten SDK {version} in '{EMSCRIPTEN_ROOT}' is older than "
             "minimum required version "
             f"{'.'.join(str(v) for v in EMSDK_MIN_VERSION)}.",
@@ -196,7 +199,7 @@ EMSCRIPTEN = Platform(
 def _check_wasi():
     wasm_ld = WASI_SDK_PATH / "bin" / "wasm-ld"
     if not wasm_ld.exists():
-        raise MissingDependency(wasm_ld, INSTALL_WASI_SDK)
+        raise MissingDependency(os.fspath(wasm_ld), INSTALL_WASI_SDK)
     wasmtime = shutil.which("wasmtime")
     if wasmtime is None:
         raise MissingDependency("wasmtime", INSTALL_WASMTIME)

